### PR TITLE
fabtests: Add option to enable OOB address exchange only

### PR DESCRIPTION
**Pull request by j-xiong**
The '-b' option enables OOB address exchange and test synchronization.
However, some tests rely on the synchronization to make progress over
certain providers. They won't work with OOB channel turned on.

Add the '-E' option that enables address exchange over the OOB channel
but keeps test synchronization going through the provider.

Signed-off-by: Jianxin Xiong <jianxin.xiong@intel.com>

### CODE REVIEW

File: fabtests/common/shared.c [hunk 1]
```diff
@@ -572,7 +572,7 @@ static int ft_init_oob(void)
 	int ret, op, err;
 	struct addrinfo *ai = NULL;
 
-	if (!(opts.options & FT_OPT_OOB_SYNC) || oob_sock != -1)
+	if ((!opts.oob_port && !(opts.options & FT_OPT_OOB_SYNC)) || oob_sock != -1)
```
Comment (shefty) [inline fabtests/common/shared.c]: Why this change?

Comment (j-xiong) [reply fabtests/common/shared.c]: The new option sets opts.oob_port w/o setting the FT_OPT_OOB_SYNC flag, but the oob channel still needs to be initialized.

Comment (shefty) [reply fabtests/common/shared.c]: Maybe we should add Ft_OPT_OOB_ADDR_EXCH rather than relying on oob_port to convey this

Comment (j-xiong) [reply fabtests/common/shared.c]: Yes, let me update the PR with that option.

File: fabtests/common/shared.c [hunk 2]
```diff
@@ -2724,6 +2726,13 @@ void ft_parse_addr_opts(int op, char *optarg, struct ft_opts *opts)
 		else
 			opts->oob_port = default_oob_port;
 		break;
+	case 'E':
```
Comment (shefty) [inline fabtests/common/shared.c]: Should the -b option fall through (like you had previously), so that setting sync also sets addr_exch?

Comment (j-xiong) [reply fabtests/common/shared.c]: The depends on if the FT_OPT_OOB_SYNC flag implies address exchange or not. Currently it is implied so there is no need to add the FT_OPT_OOB_ADDR_EXCH flag to the "-b" option. If we decide to make these two flags orthogonal then the "-b" option should have the two flags set. There are a few tests that set the FI_OPT_OOB_SYNC flag directly. They may need also to be changed if we go this path. I am fine with either choice.

Comment (shefty) [reply fabtests/common/shared.c]: This was my thought.  The -b option currently implies ADDR_EXCH, which is why I was thinking of falling through to make setting the option explicit.   From the view of the code, we keep the two flags disjoint.  It's the command line option that enables both OOB_SYNC and ADDR_EXCH.

And I also thought about the tests that are setting OOB_SYNC.  I agree those may need to be updated.  We can consider creating a new define, OOB_CTRL = OOB_SYNC | ADDR_EXCH, to mean use OOB for all control exchanges.  ?

Comment (j-xiong) [reply fabtests/common/shared.c]: OOB_CTRL sounds a good idea. I will go ahead to make the changes including the affected tests.
